sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/DeduplicateRelations.scala

```diff
@@ -95,9 +95,9 @@ object DeduplicateRelations extends Rule[LogicalPlan] {
               .flatMap(_.output).zip(newChildren.flatMap(_.output))
               .filter { case (a1, a2) => a1.exprId != a2.exprId }
           )
-          plan.withNewChildren(newChildren.toSeq).rewriteAttrs(attrMap)
+          plan.withNewChildren(newChildren.toList).rewriteAttrs(attrMap)
```

Comment:
Note to reviewers: this change is necessary to prevent tpc-ds plan stability tests to pass. It was complaining that a list of children is an `ArrayBuffer` instead of a `List`. The reason for the difference is that the new `withNewChildren` just replaces the children with the given new children and doesn't care about the exact list type.